### ISSUE TITLE
Create a dataset object similar to torch.

### ISSUE THREAD

**Issue by vandenbergheluke** [vandenbergheluke/neuralwindow#76]
Create a C version of the dataset/dataloader objects in torch that gives us a uniform interface to traverse over a dataset: https://pytorch.org/tutorials/beginner/basics/data_tutorial.html. Add features to shuffle the dataset and specify batch size. We don't want to load the entire dataset into memory.
